src/pages/Login.js

```diff
@@ -0,0 +1,232 @@
+import { colors } from '../utils/Colors';
+import LoginLogo from '../img/SwagLabs_logo.png';
+import LoginBot from '../img/Login_Bot_graphic.png';
+import { isLockedOutUser, verifyCredentials } from '../utils/Credentials';
+import ErrorMessageContainer from '../components/ErrorMessageContainer';
+import { withRouter } from 'react-router-dom';
+import { ROUTES } from '../utils/Constants';
+
+const styles = theme => ({
+  root: {
+    flexGrow: 1,
+  },
+  loginWrapper: {
+    padding: '30px 20px',
+    [ theme.breakpoints.down('xs') ]: {
+      padding: 0,
+    },
+  },
+  loginWrapperInner: {
+    border: `2px solid ${ colors.lightGray }`,
+    boxShadow: 'none',
+    maxWidth: 794,
+    margin: 'auto',
+    [ theme.breakpoints.down('xs') ]: {
+      border: 'none',
+    },
+  },
+  wrapper: {
+    paddingBottom: 30,
+    paddingTop: 30,
+  },
+  greyWrapper: {
+    backgroundColor: colors.lightGray,
+  },
+  contentWrapper: {
+    paddingLeft: 30,
+    paddingRight: 30,
+    color: colors.gray,
+    fontSize: 14,
+  },
+  logo: {
+    textAlign: 'center',
+    background: `url(${ LoginLogo }) no-repeat center center`,
+    height: 60,
+    marginTop: 30,
+  },
+  bot: {
+    textAlign: 'center',
+    background: `url(${ LoginBot }) no-repeat center center`,
+    minHeight: 260,
+    minWidth: 300,
+    [ theme.breakpoints.down('xs') ]: {
+      marginTop: 30,
+    },
+  },
+  divider: {
+    marginBottom: 20,
+  },
+});
+
+class Login extends Component {
+  static propTypes = {
+    classes: PropTypes.object.isRequired,
+    history: PropTypes.object.isRequired,
+  };
+
+  constructor(props) {
+    super(props);
+    this.state = {
+      username: '',
+      usernameError: false,
+      passwordError: false,
+      password: '',
+      error: '',
+    };
+
+    this.handlePassChange = this.handlePassChange.bind(this);
+    this.handleUserChange = this.handleUserChange.bind(this);
+    this.handleSubmit = this.handleSubmit.bind(this);
+  }
+
+  handleSubmit(evt) {
+    evt.preventDefault();
+
+    if (!this.state.username) {
+      return this.setState({
+        error: 'Username is required',
+        usernameError: true,
+        passwordError: false,
+      });
+    }
+
+    if (!this.state.password) {
+      return this.setState({
+        error: 'Password is required',
+        usernameError: false,
+        passwordError: true,
+      });
+    }
+
+    if (verifyCredentials(this.state.username, this.state.password)) {
+      // Catch our locked-out user and bail out
+      if (isLockedOutUser()) {
+        return this.setState({
+          error: 'Sorry, this user has been locked out.',
+          usernameError: false,
+          passwordError: false,
+        });
+      }
+
+      // If we're here, we have a username and password. Redirect!
+      this.setState({
+        username: '',
+        usernameError: false,
+        passwordError: false,
+        password: '',
+        error: '',
+      });
```

Comment:
Resetting the component to its initial state shouldn't be necessary since it gets unmounted when `this.props.history.push(ROUTES.INVENTORY);` is called below.

Reply:
Thanks, didn't know that